astropy/table/table.py

```diff
@@ -509,6 +509,8 @@ def _init_from_cols(self, cols):
         """Initialize table from a list of Column objects"""
 
         lengths = set(len(col) for col in cols)
+        if len(lengths) == 0:
+            raise ValueError('Cannot slice table with empty iterator')
```

Comment:
Interesting... I'm not really sure what putting this here has to do with slicing a table.  I guess, slicing a table in some cases can result in a list of columns that are used to instantiate a new `Table` object.  But this error would also be raised in the case of a non-empty slice like `table[['a', 'a']]` which means you can't create a new table with a duplicate column (should that use case be supported?  I think it is...)

It also seems a bit round-about.  Makes more sense to do this somewhere directly related to slicing.  Though in analogy to a Numpy array I wonder if slicing a table with an empty list shouldn't just return an empty table, but I'm not sure about that either.